lib/type/string.ex

```diff
@@ -36,6 +37,37 @@ defmodule Litmus.Type.String do
     end
   end
 
+  @spec convert(t, binary, map) :: {:ok, map}
+  defp convert(%__MODULE__{}, field, params) do
+    if Map.has_key?(params, field) and
+         !(is_binary(params[field]) or is_number(params[field]) or is_boolean(params[field])) do
+      {:error, "#{field} must be string"}
+    else
+      if is_binary(params[field]) do
+        {:ok, params}
+      else
+        modified_value = Kernel.inspect(params[field])
+        modified_params = Map.put(params, field, modified_value)
+        {:ok, modified_params}
+      end
+    end
```

Comment:
Instead of having nested `if` statements, we should use a `cond` statement. Also, there might be a way to split this out into three functions using pattern matching.

Reply:
I'd recommend a cond as well, and think about simplifying this logic into the main four cases:

1. params does not contain the field
2. params does contain the field and its a binary
3. params does contain the field and its a number or boolean
4. else